### PR TITLE
Add thread safety for properties

### PR DESCRIPTION
**Pull request by w4bremer**

### CODE REVIEW

File: goldenmaster/modules/testbed2_module/testbed2/generated/core/nestedstruct3interface.threadsafedecorator.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+namespace Testbed2 {
+
+class TEST_TESTBED2_EXPORT NestedStruct3InterfaceThreadSafeDecorator : public INestedStruct3Interface
+{
+protected:
+    /** 
+    * ctor
+    * @param impl The NestedStruct3Interface object to make thread safe.
+    */
+    explicit NestedStruct3InterfaceThreadSafeDecorator(INestedStruct3Interface& impl);
```
[the remark this answers is not in the document]
Comment (w4bremer) [reply goldenmaster/modules/testbed2_module/testbed2/generated/core/nestedstruct3interface.threadsafedecorator.h]: Using shared_ptr now

File: goldenmaster/modules/testbed2_module/testbed2/generated/core/nestedstruct3interface.threadsafedecorator.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+\code{.cpp}
+int main(){
+    std::unique_ptr<Testbed2::INestedStruct3Interface> testTestbed2NestedStruct3Interface = std::make_unique<Testbed2::NestedStruct3Interface>();
+    std::unique_ptr<Testbed2::INestedStruct3Interface> testTestbed2NestedStruct3InterfaceThreadSafeDecorator = std::make_unique<Test::Testbed2::NestedStruct3InterfaceThreadSafeDecorator>(*testTestbed2NestedStruct3Interface);
+
+    // Thread safe access
+    testTestbed2NestedStruct3InterfaceThreadSafeDecorator->setProp1(testTestbed2NestedStruct3InterfaceThreadSafeDecorator->getProp1());
+    testTestbed2NestedStruct3InterfaceThreadSafeDecorator->setProp2(testTestbed2NestedStruct3InterfaceThreadSafeDecorator->getProp2());
+    testTestbed2NestedStruct3InterfaceThreadSafeDecorator->setProp3(testTestbed2NestedStruct3InterfaceThreadSafeDecorator->getProp3());
+    return 0;
```
[the remark this answers is not in the document]
Comment (w4bremer) [reply goldenmaster/modules/testbed2_module/testbed2/generated/core/nestedstruct3interface.threadsafedecorator.h]: Updated docs

File: goldenmaster/modules/testbed2_module/testbed2/generated/core/manyparaminterface.threadsafedecorator.h [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    /** Guards and forwards call to ManyParamInterface implementation. */
+    void setProp4(int prop4) override;
+    /** Guards and forwards call to ManyParamInterface implementation. */
+    int getProp4() const override;
+
+    /**
+    * Access to a publisher, use it to subscribe for ManyParamInterface changes and signal emission.
+    * @return The publisher for ManyParamInterface.
+    */
+    IManyParamInterfacePublisher& _getPublisher() const override;
```
Comment (dorotaphanSiili) [inline goldenmaster/modules/testbed2_module/testbed2/generated/core/manyparaminterface.threadsafedecorator.h:91]: this is thread safe, right?
in a sense it forwards call to impl which is a shared_ptr so as long as the ThreadSafeDecorator exists it exists too
also now with this commit the publisher is thread safe
maybe the comment should also mention that it is thread safe to use it?

**Review by dorotaphanSiili — APPROVED** [review]

**Review by dorotaphanSiili — APPROVED** [review]

**Comment by w4bremer** [comment]
The macOS cmake build failure seems to be specific to the cmake conan package and not caused by the code changes. Therefore I will go on and merge the PR.

**Pull request merged**